test/utils/Fetcher.spec.ts

```diff
@@ -1,21 +1,77 @@
 import chai from 'chai'
+import mockttp = require('mockttp')
 import chaiAsPromised from 'chai-as-promised'
-import { Fetcher } from 'utils'
+import { delay, Fetcher } from 'utils'
 
 chai.use(chaiAsPromised)
 const expect = chai.expect
+const mockServer = mockttp.getLocal()
 
 describe('Fetcher', () => {
+  beforeEach(() => {
+    mockServer.start(8080)
+  })
+  afterEach(() => mockServer.stop())
 
-    it('When a timeout is set, then the fetch is aborted', async () => {
-        const fetcher = new Fetcher()
-        const fetch = fetcher.fetchBuffer('https://httpstat.us/200?sleep=10000', { timeout: '5s' })
-        await expect(fetch).to.be.rejectedWith('The user aborted a request.')
-    }).timeout('15s')
+  it('When a timeout is set, then the fetch is aborted', async () => {
+    mockServer.get('/mocked-path').thenCallback(async () => {
+      await delay('10000')
+      return {}
+    })
 
-    it('When a timeout is set but not reached, then the fetch is successful', async () => {
-        const fetcher = new Fetcher()
-        await fetcher.fetchBuffer('https://httpstat.us/200', { timeout: '5s' })
-    }).timeout('10s')
+    const fetch = new Fetcher().fetchBuffer({ url: 'http://localhost:8080/mocked-path', timeout: '5s' })
 
+    await expect(fetch).to.be.rejectedWith('The user aborted a request.')
+  }).timeout('15s')
+
+  it('When a timeout is set but not reached, then the fetch is successful', async () => {
+    await mockServer.get('/mocked-path').thenReply(200, '{"body": "matching body"}')
+
+    const fetch = new Fetcher().fetchBuffer({ url: 'http://localhost:8080/mocked-path', timeout: '5s' })
```

Comment:
is there a good reason to create an instance of that class `Fetcher`? it seems to hold no state. Could it be replaced with an exported `fetchBuffer` function?

Reply:
That's true, anyway after receiving @nchamo's feedback, I agree on having defaults configured in the constructor, so I think that I'll do that.